update Numark Mixtrack Pro mapping

qguv opened the pull request:
Fixes some broken mappings and exposes some more Mixxx functionality to the controller user, see comment at the top of [the script](https://github.com/qguv/mixxx/blob/numark-mixtrack-pro/res/controllers/Numark-Mixtrack-Pro-scripts.js). Thanks!

qguv committed:
update numark mixtrack pro mapping

uklotzde commented:
Welcome and thanks for contributing. As a first-time contributor we have to ask you to sign our contributor agreement:

https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ

Please comment here when done.

TODO:
- Rebase your branch on 2.3, force push and then change the target branch to 2.3. This ensures that the updated mapping will become available in 2.3.x when ready
- Fix the pre-commit issues code style issues
- Update the manual page for this controller and file a PR for the manual. Also use the 2.3 branch as a base. When done please link the manual PR in the description of this mapping PR. Changes to mappings are only merged together with the corresponding updated manual page.

@Swiftb0y @Holzhaus Your turn ;)

uklotzde commented:
I have deliberately assigned the 2.3.1 milestone to mark this PR as optional for 2.3.0. Depending on the progress it may as well be merged before 2.3.0 is released.

Holzhaus commented on res/controllers/Numark Mixtrack Pro.midi.xml:
```diff
@@ -1,846 +1,747 @@
+        <name>Numark MixTrack Pro</name>
+        <author>Matteo (matteo@magm3.com), James Ralston, and D. J. Freije (dario2004@gmail.com)</author>
+        <description>version v1.2 w/brake, backspin, blink beat Leds.</description>
+        <forums>http://mixxx.org/forums/viewtopic.php?f=7&amp;t=3828</forums>
+        <wiki>http://www.mixxx.org/wiki/doku.php/numark_mixtrack_pro</wiki>
```
please revert this change.

qguv committed:
fix style issues

qguv committed:
fix wording in comment

qguv committed:
revert metadata change

qguv committed:
update version of midi xml mapping

qguv commented:
Hi all, thanks for the really fast review.

- [x] I've reverted the changes to the XML metadata section of the midi mapping
- [x] I've rebased onto the `2.3` branch and changed the PR's target branch
- [x] I've made the automated linter reasonably happy
- [x] I've updated the manual and created a PR at mixxxdj/manual#390
- [x] I've signed the contributor agreement

uklotzde commented:
Did you install and run the pre-commit hook locally as described [in the Wiki](https://github.com/mixxxdj/mixxx/wiki/Coding-Guidelines#pre-commit)?

wtfrank commented:
I've installed the updated mixtrack pro mapping and can confirm it gets rid of the error at startup to do with flanger assignment.

Unrelated to this change really, but one thing I've wondered about the mixtrack pro mapping is the EQ Kill buttons in this photo.
![image](https://user-images.githubusercontent.com/1202593/118391433-275aa400-b62c-11eb-95ad-0f304ff58c43.png)

These are mapped in both this version and the old version as cue buttons. But this functionality isn't documented inside the mapping file. I was wondering whether there were different versions of the mixtrack pro - some with buttons marked as cues and some as eq kills - or whether it was just an undocumented decision to map them to cues that one of the early authors of the mapping made.

qguv committed:
fix style warnings

uklotzde commented:
@wtfrank Thanks for testing, much appreciated! We need users with the controller at hand to do this.

uklotzde commented:
Unfortunately, the pre-commit check still complains about the indentations.

qguv committed:
eslint: autofix indentation

qguv committed:
eslint: autofix all

uklotzde reviewed (approved):
@Holzhaus @Swiftb0y What do you think? Since this fixes a partially broken mapping I suggest to merge it asap.

Swiftb0y commented:
Sorry, I don't have the time right now to review this mapping.

Holzhaus commented:
I left some comments on https://github.com/mixxxdj/manual/pull/390.

uklotzde commented:
@qguv Any updates on the manual PR? It would be nice to include the fixed mapping in 2.3.0.

qguv commented:
Hi @uklotzde , I'm not able to contribute on weekdays. I'll take a look this weekend.

uklotzde commented:
Any progress on this? Rescheduling to 2.3.2.
